### DISCUSSION TITLE
Microfrontends + Astro

### DISCUSSION THREAD

**Discussion by roblevintennis** [AgnosticUI/agnosticui#129] [Ideas] [answered]
While on [The Function Call podcast](https://youtu.be/UoHamm66UBA?t=1553) Austin had an interesting idea that there may be a use case for AgnosticUI for creating microfrontends that are stitched together into a fully static site e.g. using something like Astro.build. As this would prevent the typical complaint about shipping 2+ frameworks to the user since it's all getting compiled to static site, this might be an interesting use case value prop for exploiting AgnosticUI's ability to use "1 stylesheet many frameworks" e.g. we can imagine a Svelte microfrontend, a Vue microfrontend, and a React microfrontend all getting &ldquo;stitched together&rdquo; by Astro but sharing the same branded theme.

According to Astro docs [framework-components](https://docs.astro.build/en/core-concepts/framework-components/) page (See **Mixing Frameworks**) it IS possible.

This merits some exploration:
https://github.com/AgnosticUI/agnosticui/issues/128

**Comment by roblevintennis** [comment, marked as answer]
https://github.com/AgnosticUI/AstroAgnosticUIStarter

**Reply by robsongajunior** [reply]
Where is the config for moule federeation?
If you have the repeated css on the two files will keep duplicating

**Comment by JayMGurav** [comment]
Interesting! how is this in comparison with [Webpack's module-federation](https://webpack.js.org/concepts/module-federation)

**Reply by roblevintennis** [reply]
I'm not really sure...once I started hearing about module federation I'd generally started to move away from Webpack due to fatigue and the speed and seemingly improved dux of other alternatives namely Vite.

That said there's typically alternatives to all solutions and so it's probably an "it depends" for any comparison but I don't have much useful to provide in terms of a comparison.

**Comment by roblevintennis** [comment]
I'm closing out legacy discussions as the v2 rewrite is almost ready and is a complete rewrite overhaul so discussions are no longer relevant at this point.
